yarn-project/cli/src/index.ts

```diff
@@ -377,13 +385,19 @@ export function getProgram(log: LogFn, debugLogger: DebugLogger): Command {
       const wallet = await getSchnorrAccount(client, privateKey, privateKey, accountCreationSalt).getWallet();
       const contract = await Contract.at(contractAddress, contractAbi, wallet);
       const tx = contract.methods[functionName](...functionArgs).send();
-      await tx.wait();
-      log('\nTransaction has been mined');
-      const receipt = await tx.getReceipt();
       log(`Transaction hash: ${(await tx.getTxHash()).toString()}`);
-      log(`Status: ${receipt.status}\n`);
-      log(`Block number: ${receipt.blockNumber}`);
-      log(`Block hash: ${receipt.blockHash?.toString('hex')}`);
+      if (options.wait) {
+        await tx.wait();
+
+        log('Transaction has been mined');
+
+        const receipt = await tx.getReceipt();
+        log(`Status: ${receipt.status}\n`);
+        log(`Block number: ${receipt.blockNumber}`);
+        log(`Block hash: ${receipt.blockHash?.toString('hex')}`);
+      } else {
+        log('\nTransaction pending. Check status with get-tx-receipt');
```

Comment:
You will need to provide the transaction hash in order for them to use the receipt.